develop/tutorials/articles/290-modularity-and-osgi/01-the-benefits-of-modularity.markdown

```diff
@@ -0,0 +1,167 @@
+    controls. 
+
+![Figure 1: The Apollo spacecraft's modules collectively took astronauts to the moon's surface and back to Earth.](../../images/modularity_apollo_spacecraft_diagram.png)
+
+The spacecraft and its modules exemplified these modularity characteristics: 
+
+-   **Distinct functionality**: Each module provides a distinct function;
+    modules can be combined to provide an entirely new collective function. 
+
+    The Apollo spacecraft's modules were grouped together for a distinct
+    collective function: take astronauts from the Earth's atmospheric rim, to
+    the moon's surface, and back to Earth. The previous list identifies each
+    module's distinct function.
+
+-   **Dependencies**: Modules can require capabilities other modules satisfy. 
+
+    The Apollo modules had these dependencies:
+    - Lunar Module depended on the Service Module to get near the moon.
+    - Command Module depended on the Service Module for power and oxygen.
+    - Service Module depended on the Command Module for instruction.
+
+-   **Encapsulation**: Modules hide their implementation details but publicly
+    define their capabilities and interfaces. 
+
+    Each Apollo module was commissioned with a contract defining its
+    capabilities and interface, while each module's details were encapsulated
+    (hidden) from other modules. NASA integrated the modules based on their
+    interfaces. 
+
+-   **Reusability**: A module can be applied to different scenarios.
+
+    The Command Module's structure and design were reusable. NASA used different
+    versions of the Command Module, for example,  throughout the Apollo program,
+    and in the Gemini Program, which focused on Earth orbit. 
+
+NASA used modularity to successfully complete over a dozen missions to the moon.
+Can modularity benefit software too? Yes! The following sections show you how:
+
+- [Modularity benefits for software](#modularity-benefits-for-software)
+- [Example: How to design a modular application](#example-designing-a-modular-application)
+
+## Modularity Benefits for Software [](id=modularity-benefits-for-software)
+
+Java applications have predominantly been monolithic: they're developed in large
+code bases. In a monolith, it's difficult to avoid tight coupling of classes.
+Modular application design, conversely, facilitate loose coupling, making the
+code easier to maintain. It's much easier and more fun to develop small amounts
+of cohesive code in modules. Here are some key benefits of developing modular
+software. 
+
+### Distinct Functionality [](id=distinct-functionality)
+
+It's natural to focus on developing one piece of software at a time. In a
+module, you work on a small set of classes to define and implement the module's
+function. Keeping scope small facilitates writing high quality, elegant code.
+The more cohesive the code, the easier it is to test, debug, and maintain.
+Modules can be combined to provide a new function, indistinguishable from each
```

Comment:
When modules are combined, the resulting artifact IS distinguishable from each individual module, right? The same modules can even be composed in different ways to build different solutions.

Reply:
Right, right ... it is indeed distinguishable. Thanks for catching that.